packages/driver/cypress/integration/commands/navigation_spec.js

```diff
@@ -2307,6 +2306,32 @@ describe('src/cy/commands/navigation', () => {
         return null
       })
     })
+
+    it('does not wait for stability at the end of the command queue when not stable with experimentalSessionAndOrigin', (done) => {
+      const onLoad = cy.spy()
+
+      cy
+      .visit('/fixtures/generic.html')
+      .then((win) => {
+        cy.on('window:load', onLoad)
+
+        cy.on('command:queue:end', () => {
+          expect(onLoad).not.have.been.called
+          done()
+        })
+
+        cy.on('command:queue:before:end', () => {
+        // force us to become unstable immediately
+        // else the beforeunload event fires at the end
+        // of the tick which is too late
```

Comment:
```suggestion
          // force us to become unstable immediately
          // else the beforeunload event fires at the end
          // of the tick which is too late
```

Reply:
Addressed in https://github.com/cypress-io/cypress/pull/21137/commits/c486640a7daee8d8e2f42ac7829d073b42ed6bbf